src/wmtk/Mesh.hpp

```diff
@@ -67,9 +67,10 @@ class Mesh : public std::enable_shared_from_this<Mesh>
     /**
      * Generate a vector of Tuples from global vertex/edge/triangle/tetrahedron index
      * @param type the type of tuple, can be vertex/edge/triangle/tetrahedron
+     * @param include_deleted if true returns also the deleted tuples (default false)
      * @return vector of Tuples referring to each type
      */
-    std::vector<Tuple> get_all(PrimitiveType type) const;
+    std::vector<Tuple> get_all(PrimitiveType type, const double include_deleted = false) const;
```

Comment:
bool.

Reply:
Also I'm not sure if I agree with this API - it will almost certainly break and SHOULD break in debug mode because in theory every access to a tuple should validate that the tuple is valid and the tuple belonging to an active cell should be active. 

The point is to allow paraviewwriter to figure out which indices are active or not, and this seems like something that could be read off by the paraviewwriter using the flag_attribute_handles() and the AttributeManager object directly using indices. Rather than introduce an implicit index-revealing semantic in the `get_all` function I'd rather let the mesh writers have access to indices